unlock-app/src/__tests__/services/web3Service.test.js

```diff
@@ -1073,4 +1073,24 @@ describe('Web3Service', () => {
       expect(web3Service.getTransaction).toHaveBeenCalledWith(transactionHash)
     })
   })
+
+  describe('inputsHandlers', () => {
+    describe('createLock', () => {
+      it('should emit lock.updated with correctly typed values', async done => {
+        expect.assertions(2)
+        const params = {
+          _expirationDuration: '7',
+          _maxNumberOfKeys: '5',
+          _keyPrice: '0.05',
+        }
+        web3Service.on('lock.updated', (newLockAddress, update) => {
+          expect(update.expirationDuration).toBe(7)
+          expect(update.maxNumberOfKeys).toBe(5)
+          done()
+        })
+
+        await web3Service.inputsHandlers.createLock('0x123', '0x456', params)
```

Comment:
skip the async/await part.

Reply:
since you have a `done` param the test will not finish until the done is called. using async/await will mnake it finish as soon as the promise resolves.

Reply:
Let me know if that solves it!

Reply:
Same issue as before, unfortunately. Fails with:
```
 FAIL  src/__tests__/services/web3Service.test.js (8.556s)
  ● Web3Service › inputsHandlers › createLock › should emit lock.updated with correctly typed values

    Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.

      1077 |   describe('inputsHandlers', () => {
      1078 |     describe('createLock', () => {
    > 1079 |       it('should emit lock.updated with correctly typed values', done => {
           |       ^
      1080 |         expect.assertions(2)
      1081 |         const params = {
      1082 |           _expirationDuration: '7',

      at Spec (../node_modules/jest-jasmine2/build/jasmine/Spec.js:92:20)
      at Suite.it (src/__tests__/services/web3Service.test.js:1079:7)
      at Suite.describe (src/__tests__/services/web3Service.test.js:1078:5)
      at Suite.describe (src/__tests__/services/web3Service.test.js:1077:3)
      at Object.describe (src/__tests__/services/web3Service.test.js:95:1)

  ● Web3Service › inputsHandlers › createLock › should emit lock.updated with correctly typed values

    expect.assertions(2)

    Expected two assertions to be called but received zero assertion calls.

      1078 |     describe('createLock', () => {
      1079 |       it('should emit lock.updated with correctly typed values', done => {
    > 1080 |         expect.assertions(2)
           |                ^
      1081 |         const params = {
      1082 |           _expirationDuration: '7',
      1083 |           _maxNumberOfKeys: '5',

      at Object.assertions (src/__tests__/services/web3Service.test.js:1080:16)
```

Reply:
I wonder if the issue is not `generateLockAddress` failing in some way?

Reply:
Ah, let me mock that out and see what happens